### ISSUE TITLE
How do N'Ko counter styles work?

### ISSUE THREAD

**Issue by r12a** [w3c/afrlreq#26]
[1] In Wikipedia i see the following list, using N'Ko digits for counters and with a hyphen for a separator.

<img width="407" alt="Screenshot 2023-03-08 at 10 47 18" src="https://user-images.githubusercontent.com/4839211/223696024-ba56bbfa-e8a1-4362-a440-1e8161c400d9.png">

Is this typical? If so, I'll document it in our [Ready-made Counter Styles doc](https://www.w3.org/TR/predefined-counter-styles/#adlam-styles).

[2] I also see another list which appears to have ordinal counters.

<img width="591" alt="Screenshot 2023-03-08 at 10 48 04" src="https://user-images.githubusercontent.com/4839211/223696321-461f18a1-2669-4efb-a00e-bf841f7ebfef.png">

Is this another typical style that we should document?  I notice that sometimes the dot is missing, and i'm guessing that that's a typo.  It also happens in item 11 - i'm guessing that the sequence would continue using dots like this:
<img width="700" alt="Screenshot 2023-03-08 at 11 14 55" src="https://user-images.githubusercontent.com/4839211/223699321-d08adb73-7b29-4fe5-aafe-2b1a1d792d17.png">
Is that correct?

[3] Does N'Ko use alphabetic counter styles that we should also add?  If so, can someone provide the ordered list of letters that would be used, and indicate what happens after the list becomes longer than the number of letters available (usually that would be ...z aa bb ...) (A scan of an example would be useful.)

**Comment by NeilSureshPatel** [comment]
I have asked Coleman Donaldson about the usage of ordinals and whether alphabetic counters are used. In regards to how the ordinals continue after the number 10, the mark continues to appear as you suggest. There are some missing dots in the examples below but I think that they are typos.

![CCI_000005](https://user-images.githubusercontent.com/46502494/224066564-45176913-0fdb-455d-b611-f8eaefc64468.jpg)
![CCI_000006](https://user-images.githubusercontent.com/46502494/224066614-4e59922c-3a1e-4c24-bceb-0c23d1f8d64c.jpg)

**Comment by r12a** [comment]
@NeilSureshPatel This is really helpful, especially to have the pictures.

Any problem with me adding the pictures to our type-samples repo? https://github.com/w3c/type-samples

So do we think that the ordinal approach is the default way to number lists, or is it an alternative ?

**Comment by NeilSureshPatel** [comment]
@r12a. No problem adding the images to the type-samples repo.

I am not sure if we can say if the ordinal style is the default. The same book also has a list without them. I am waiting to hear back on it. I am wondering if the method depends on the type of thing being listed. 

![image_000006](https://user-images.githubusercontent.com/46502494/224120611-9493181f-ded2-435d-b4c5-bce074d63a00.jpg)
![CCI_000007](https://user-images.githubusercontent.com/46502494/224120633-d3c35cc6-87b5-445f-9a65-76be1e00c7f2.jpg)

**Comment by r12a** [comment]
Ah, this set seems to confirm the use of the hyphen as the separator. That's useful.

**Comment by r12a** [comment]
@NeilSureshPatel could you let me know what this book is, since i'd like to identify the source in the type samples repo.

**Comment by NeilSureshPatel** [comment]
The book is titled, "ߒߞߏ ߕߎ߬ߜߍ߬ߕߎ߬ߜߍ߬ߟߌ ߣߴߊ߬ ߢߎߡߍߙߋ߲ߦߊߟߌ"  It is written by Solo Farabado Cisse (ߛߏ߬ߟߏ߫ ߝߙߊ߬ߓߊ-ߘߐ߫ ߛߌ߬ߛߋ߫)

I am flipping through another book right now and I am seeing both ordinal and non-ordinal lists.  Both types of lists use a hyphen as a separator.

**Comment by DD-fwd** [comment]
@r12a @NeilSureshPatel both have it correct. Here is my comprehension: cardinal numbers do not have any dot, similar to image 3 up to 11. When cardinals are used in listing, the separator is used (not sure if other type of separators are used). Ordinal numbers have dot on the bottom of the last digit.

Question 1: hyphen separator should be considered typical.
Question 2: ordinal counters are also typical.
Question 3: ordinal have dot on the bottom of the last digit (most left digit since this is RTL).

 I will validate my understanding further and update on the posting.

**Comment by r12a** [comment]
@NeilSureshPatel @DD-fwd I have just published a first draft of N'Ko Layout Requirements at and mentioned this information at https://w3c.github.io/afrlreq/nko/#h_counters

Comments welcome.  (In fact comments on any part of that document are welcome, but please raise separate issues for each topic that is not counter-styles related.)

(fwiw, it's also at https://r12a.github.io/scripts/nkoo/nqo.html#lists, since i'm creating the initial versions of these new layout req documents by extracting information from my own materials, in order to get them up and running quickly.)

**Comment by NeilSureshPatel** [comment]
I heard back regarding the choice of ordinals vs cardinal lists. Ordinal lists tend to get used when showing hierarchy of the listed items or if the author feels that the list would read more naturally as ordinals. It sounds like both methods are typical. There is also an alphabetical counter that uses the following order (RTL). After the vowels it follows the standard dictionary order.


ߊ ߋ ߌ ߍ ߎ ߏ ߐ ߒ ߓ ߔ ߕ ߖ ߗ ߘ ߙ ߛ ߜ ߝ ߞ ߟ ߡ ߧ ߤ ߥ ߦ

Here is an example. I don't have one that goes through the entire alphabet, but this illustrates it use.

![image](https://user-images.githubusercontent.com/46502494/224361450-adb3d637-5617-4b16-8fca-c26233969764.png)

**Comment by r12a** [comment]
Ah, that's interesting.  The numeric counters seem to use a hyphen as a separator, but the alphabetic counters look like <span class="codepoint" translate="no"><bdi lang="nqo" dir="rtl">&#x07FA;</bdi> [<span class="uname">U+07FA NKO LAJANYALAN</span>]</span>.  I was wondering whether that would crop up somewhere.

I'll add this information to the layout requirements drafts, thanks.

**Comment by r12a** [comment]
I'll also add these to our Ready-made Counter Styles doc at https://www.w3.org/TR/predefined-counter-styles/, although i'm not sure how to create a pattern for the ordinal counters, given the dot placements.

**Comment by NeilSureshPatel** [comment]
> Ah, that's interesting. The numeric counters seem to use a hyphen as a separator, but the alphabetic counters look like ߺ [U+07FA NKO LAJANYALAN]. I was wondering whether that would crop up somewhere.
> 
> I'll add this information to the layout requirements drafts, thanks.

The use of the Lanjanyalan though is interesting but might not be a consistent rule. Here is another example with hyphens. Coleman and I were discussing it a bit. That first image was from an older book. The one below is from a newer book.

![PXL_20230310_160045174](https://user-images.githubusercontent.com/46502494/224364394-4e2fc7b4-1b58-4f92-a9f5-5ddfad8586c4.jpg)

**Comment by NeilSureshPatel** [comment]
> I'll also add these to our Ready-made Counter Styles doc at https://www.w3.org/TR/predefined-counter-styles/, although i'm not sure how to create a pattern for the ordinal counters, given the dot placements.

Is the problem related to fact that the ߁ has the dot above? or that the dot is on the last number in the sequence?

**Comment by r12a** [comment]
Yeah, there are a couple of problems, given the regularity usually expected by the pattern syntax: one is that dots only appear on the last digit in the sequence, and the other is that the dot goes above if it happens to be a ߁.  The usual syntax for a numeric pattern would be pretty simple, eg:

@counter-style [arabic-indic](https://www.w3.org/TR/predefined-counter-styles/#arabic-indic) {
system: numeric;
symbols: '\660' '\661' '\662' '\663' '\664' '\665' '\666' '\667' '\668' '\669';
/* symbols: '٠' '١' '٢' '٣' '٤' '٥' '٦' '٧' '٨' '٩'; */
}

**Comment by DD-fwd** [comment]
> I heard back regarding the choice of ordinals vs cardinal lists. Ordinal lists tend to get used when showing hierarchy of the listed items or if the author feels that the list would read more naturally as ordinals. It sounds like both methods are typical. There is also an alphabetical counter that uses the following order (RTL). After the vowels it follows the standard dictionary order.
> 
> ߊ ߋ ߌ ߍ ߎ ߏ ߐ ߒ ߓ ߔ ߕ ߖ ߗ ߘ ߙ ߛ ߜ ߝ ߞ ߟ ߡ ߧ ߤ ߥ ߦ
> 
> Here is an example. I don't have one that goes through the entire alphabet, but this illustrates it use.
> 
> ![image](https://user-images.githubusercontent.com/46502494/224361450-adb3d637-5617-4b16-8fca-c26233969764.png)

Good you have ordinal vs cardinal list clarified. The issue with ordinal 1 pattern not similar to others may be related to the pronunciation (which is very different compared to others).
